lib/engine/game/g_1825/step/track_and_token.rb

```diff
@@ -74,6 +74,10 @@ def lay_tile_action(action, entity: nil, spender: nil)
             if tile_lay[:cannot_reuse_same_hex] && @round.laid_hexes.include?(action.hex)
               raise GameError, "#{action.hex.id} cannot be layed as this hex was already layed on this turn"
             end
+            if (tile.name == '888' || tile.name == '887') &&
+               (action.hex.id == 'Q13') && (entity.name != 'NS')
+              raise GameError, 'Only the NS can place tiles 887/888 on Q13'
+            end
```

Comment:
This works, but it will show 888 and 887 as possible upgrades for Q13 for the other companies. This should probably go into @game.upgrades_to?

Reply:
I've tried that, but on slapping in some debugging output, upgrades_to never seems to be called in 1825 with selected_company other than nil. I guess I could work out everywhere it gets called from and which of the base methods I need to replace but it seems like that might make the implementation otherwise more fragile because if those base methods change, g_1825 won't keep up.

(My terminology may be rather confused; sorry)

Reply:
Yeah, that's unfortunate. Never mind then.

Reply:
I think you can put this in upgradable_tiles, which does get the current entity as a parameter. It is the ultimate cause of upgrades_to?

Reply:
Your call

Reply:
I implemented this - hence the appearance of a protect_ns method in the update I'm about to push - but the result is that when you go to upgrade Q13, the 887/888s don't appear and there is no feedback as to why; you just have to know that rule in variant DB1 or infer that it was necessary. This strikes me as worse than being able to try it and getting a clear message as to why it's not permitted. Therefore, I unimplemented it.